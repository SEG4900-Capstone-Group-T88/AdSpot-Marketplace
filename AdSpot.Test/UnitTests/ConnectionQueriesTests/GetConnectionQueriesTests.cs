using AdSpot.Models;

namespace AdSpot.Test.UnitTests.ConnectionQueriesTests;

[Collection("adspot-inmemory-db")]
public class GetConnectionQueriesTests
{
    public const string GetConnectionQuery = """
        query GetConnection($userId: Int!, $platformId: Int!) {
            connection(userId: $userId, platformId: $platformId) {
                userId
                platformId
            }
        }
        """;

    public const string GetConnectionsQuery = """
        query GetConnections($userId: Int!) {
            connections(userId: $userId) {
                userId
                platformId
            }
        }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task GetConnectionSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(
            scope =>
            {
                var context = scope.ServiceProvider.GetRequiredService<AdSpotDbContext>();
                context.Connections.Add(
                    new Connection
                    {
                        UserId = TestDatabase.TestUser.UserId,
                        PlatformId = TestDatabase.Platforms.First().PlatformId,
                        Handle = "TestAccountHandle",
                        Token = "TestApiToken"
                    }
                );
                context.SaveChanges();
            },
            b =>
                b.SetQuery(GetConnectionQuery)
                    .SetVariableValue(
                        "userId",
                        TestDatabase.TestUser.UserId
                    )
                    .SetVariableValue(
                        "platformId",
                        TestDatabase.Platforms.First().PlatformId
                    )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task GetConnectionEmpty()
    {
        var result = await TestServices.ExecuteRequestAsync(
            b =>
                b.SetQuery(GetConnectionQuery)
                    .SetVariableValue(
                        "userId",
                        TestDatabase.TestUser.UserId
                    )
                    .SetVariableValue(
                        "platformId",
                        TestDatabase.Platforms.First().PlatformId
                    )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task GetConnectionsSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(
            scope =>
            {
                var context = scope.ServiceProvider.GetRequiredService<AdSpotDbContext>();
                context.Connections.Add(
                    new Connection
                    {
                        UserId = TestDatabase.TestUser.UserId,
                        PlatformId = TestDatabase.Platforms.First().PlatformId,
                        Handle = "TestAccountHandle",
                        Token = "TestApiToken"
                    }
                );
                context.SaveChanges();
            },
            b =>
                b.SetQuery(GetConnectionsQuery)
                    .SetVariableValue(
                        "userId",
                        TestDatabase.TestUser.UserId
                    )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task GetConnectionsEmpty()
    {
        var result = await TestServices.ExecuteRequestAsync(
            b =>
                b.SetQuery(GetConnectionsQuery)
                    .SetVariableValue(
                        "userId",
                        TestDatabase.TestUser.UserId
                    )
        );

        result.MatchSnapshot();
    }
}