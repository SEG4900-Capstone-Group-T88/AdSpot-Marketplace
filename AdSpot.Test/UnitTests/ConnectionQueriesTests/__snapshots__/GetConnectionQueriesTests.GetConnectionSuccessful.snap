{
  "data": {
    "connection": [
      {
        "userId": 1,
        "platformId": 1
      }
    ]
  }
}